app/src/org/commcare/android/view/SquareButton.java

```diff
@@ -27,7 +27,12 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         int w = MeasureSpec.getSize(widthMeasureSpec);
         int h = MeasureSpec.getSize(heightMeasureSpec);
 
-        w = Math.max(w, h);
+        if(w == 0 || h == 0) { // this can happen in the home screen gridView, and then we'll have an invisible view
```

Comment:
I'm sort of irrationally picky about this, but can you put comments on their own lines, and inside of the conditional body that they correspond to?